acceptance/tests/lookup/merge_strategies.rb

```diff
@@ -0,0 +1,172 @@
+test_name 'C99903: merge strategies' do
+  require 'puppet/acceptance/environment_utils.rb'
+  extend Puppet::Acceptance::EnvironmentUtils
+
+  app_type        = File.basename(__FILE__, '.*')
+  tmp_environment = mk_tmp_environment_with_teardown(master, app_type)
+  fq_tmp_environmentpath  = "#{environmentpath}/#{tmp_environment}"
+  tmp_environment2 = mk_tmp_environment_with_teardown(master, app_type)
+  fq_tmp_environmentpath2  = "#{environmentpath}/#{tmp_environment2}"
+
+  confdir = master.puppet('master')['confdir']
+  codedir = master.puppet('master')['codedir']
+
+  teardown do
+    step "remove global hiera.yaml" do
+      on(master, "rm #{confdir}/hiera.yaml")
```

Comment:
The global hiera.yaml is installed by packages, so we shouldn't remove it, and the test should use a temp hiera.yaml instead (or backup/restore).